lib/mysql2/client.rb

```diff
@@ -82,8 +82,18 @@ def self.default_query_options
     if Thread.respond_to?(:handle_interrupt)
       require 'timeout'
 
+      # Timeout::ExitException was removed in Ruby 2.3.0, 2.2.3, and 2.1.8,
+      # but is present in earlier 2.1.x and 2.2.x, so we provide a shim.
+      #
+      # Set our own constant instead of modifying core ::Timeout.
+      TimeoutError = if defined?(::Timeout::ExitException)
```

Comment:
Idiomatic to use a constant when it won't be reassigned and we want lexical lookup from an instance method. If this is a value we needed to configure, using a class or instance attribute would make sense.